src/access/profiling/manager.py

```diff
@@ -49,15 +50,46 @@ def parse(self) -> xr.Dataset:
         )
 
 
+class ProfilingExperimentStatus(Enum):
+    """Enumeration representing the status of a profiling experiment."""
+
+    NEW = 1
+    RUNNING = 2
+    DONE = 3
+
+
+class ProfilingExperiment:
+    """Represents a profiling experiment.
+
+    Args:
+        path (Path): Path to the experiment directory.
+    """
+
+    path: Path  # Path to the experiment directory
+    status: ProfilingExperimentStatus = ProfilingExperimentStatus.NEW  # Status of the experiment
+
+    def __init__(self, path: Path) -> None:
+        self.path = path
+
+
 class ProfilingManager(ABC):
     """Abstract base class to handle profiling data and workflows.
 
     This high-level class defines methods to parse different types of profiling data. Currently,
     it supports parsing and plotting scaling data.
+
+    Args:
+        work_dir (Path): Path to directory used to generate and run profiling experiments.
     """
 
+    work_dir: Path  # Path to directory used to generate and run profiling experiments.
+    experiments: dict[str, ProfilingExperiment] = {}  # Dictionary storing ProfilingExperiment instances.
```

Comment:
Sorry - what does this annotation `[str, ProfilingExperiment]` mean - is it that the key is one of `str` or `ProfilingExperiment`?

Reply:
Learnt something :)